src/core/Akka.Remote/Endpoint.cs

```diff
@@ -411,147 +414,174 @@ protected override void PostRestart(Exception reason)
+            });
+            Receive<AttemptSysMsgRedelivery>(sysmsg =>
+            {
+                if (UidConfirmed) ResendAll();
+            });
+            Receive<Terminated>(terminated =>
+            {
+                _currentHandle = null;
+                Context.Parent.Tell(new EndpointWriter.StoppedReading(Self));
+                if (_resendBuffer.NonAcked.Any() || _resendBuffer.Nacked.Any())
+                    Context.System.Scheduler.ScheduleTellOnce(_settings.SysResendTimeout, Self,
+                        new AttemptSysMsgRedelivery(), Self);
+                GoToIdle();
+            });
+            Receive<GotUid>(g =>
+            {
+                _bailoutAt = null;
+                Context.Parent.Tell(g);
+                //New system that has the same address as the old - need to start from fresh state
+                UidConfirmed = true;
+                if (Uid.HasValue && Uid.Value != g.Uid) Reset();
+                Uid = g.Uid;
+                ResendAll();
+            });
+            Receive<EndpointWriter.StopReading>(stopped =>
+            {
+                _writer.Forward(stopped); //forward the request
+            });
+        }
 
-                        ResendNacked();
-                    }
-                })
-                .With<AttemptSysMsgRedelivery>(sysmsg =>
-                {
-                    if (UidConfirmed) ResendAll();
-                })
-                .With<Terminated>(terminated =>
-                {
-                    _currentHandle = null;
-                    Context.Parent.Tell(new EndpointWriter.StoppedReading(Self));
-                    if (_resendBuffer.NonAcked.Count > 0 || _resendBuffer.Nacked.Count > 0)
-                        Context.System.Scheduler.ScheduleTellOnce(_settings.SysResendTimeout, Self, 
-                            new AttemptSysMsgRedelivery(), Self);
-                    Context.Become(Idle);
-                })
-                .With<GotUid>(g =>
-                {
-                    Context.Parent.Tell(g);
-                    //New system that has the same address as the old - need to start from fresh state
-                    UidConfirmed = true;
-                    if (Uid.HasValue && Uid.Value != g.Uid) Reset();
-                    else UnstashAcks();
-                    Uid = _refuseUid;
-                })
-                .With<EndpointWriter.StopReading>(stopped =>
-                {
-                    _writer.Forward(stopped); //forward the request
-                });
+        private void GoToIdle()
+        {
+            if (_bufferWasInUse && _maxSilenceTimer == null)
+                _maxSilenceTimer =
+                    Context.System.Scheduler.ScheduleTellOnceCancelable(_settings.QuarantineSilentSystemTimeout, Self,
+                        TooLongIdle.Instance, Self);
+            Become(IdleBehavior);
         }
 
-        protected void Gated(object message)
+        private void GoToActive()
```

Comment:
Transition method that takes us out of a `Gated` or `Idle` state and back into an active state where we're receiving and sending messages normally.

Cancels the `QuarantineSilenceTimeout` timer.